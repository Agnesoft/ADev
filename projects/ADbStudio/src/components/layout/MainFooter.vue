<template>
    <footer class="main-footer">
        <ul class="copyright">
            <li>Â© 2020 ADbStudio</li>
            <li><a href="https://www.agnesoft.com">Agnesoft</a></li>
        </ul>
    </footer>
</template>

<script>
export default {
    name: "MainFooter",
}
</script>

<style scoped>
    .main-footer{
        background-color: var(--dark-color);
        color: var(--light-color);
    }
    .copyright{
        position: relative;
        display: flex;
        justify-content: space-between;
        padding: 0.5rem 1rem;
        margin: 0.5rem;
    }

    .copyright:before{ 
        display: block;
        content: "";
        position: absolute;
        top: -0.5rem;
        left: 0;
        height: 1px;
        width: 100%;
        background: var(--light-color);
        background: var(--gradient-underline);
    }
    .copyright a:hover{
        opacity: 0.7;
    }
    .main-footer a{
        color: var(--light-color);
    }
</style>
