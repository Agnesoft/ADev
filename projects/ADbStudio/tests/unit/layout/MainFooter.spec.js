import { shallowMount  } from '@vue/test-utils';
import MainFooter from '@/components/layout/MainFooter.vue';

import { localVue, i18n } from '../TestUtils'


describe('MainHeader',() => {
    let wrapper
    beforeAll(() => {
        wrapper = shallowMount(MainFooter, {
            localVue,
            i18n,
          });
    })

    it('renders copyright', () => {
        expect(wrapper.find(".copyright").html()).toContain("Â© 2020 ADbStudio");
        expect(wrapper.find(".copyright").html()).toContain('<a href="https://www.agnesoft.com">Agnesoft</a>');
    })
})