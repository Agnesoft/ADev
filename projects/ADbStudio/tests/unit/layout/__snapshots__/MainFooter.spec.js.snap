// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainHeader renders correctly (snapshot) 1`] = `
<footer
  class="main-footer"
>
  <ul
    class="copyright"
  >
    <li>
      Â© 2020 ADbStudio
    </li>
     
    <li>
      <a
        href="https://www.agnesoft.com"
      >
        Agnesoft
      </a>
    </li>
  </ul>
</footer>
`;
